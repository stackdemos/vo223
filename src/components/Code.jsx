import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Code.scss';

export function Code(props, ...args) {
    if (typeof props === 'string') {
        return <Code>{[props, ...args].join('\n')}</Code>;
    }
    const {className, inline, ...rest} = props;
    const componentClassName = classNames(
        'code',
        className,
        {
            'code-inline': inline
        }
    );

    return <code className={componentClassName} {...rest} />;
}

Code.propTypes = {
    className: PropTypes.string,
    inline: PropTypes.bool
};

Code.defaultProps = {
    className: undefined,
    inline: false
};

export default Code;
