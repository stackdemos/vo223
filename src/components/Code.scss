@import 'vars';

.code {
    display: block;
    padding: 20px;
    margin: 10px 80px;
    background: $theme-code-background;
    white-space: pre;
}

.code-inline {
    display: inline;
    padding: 3px;
    margin: 3px;
}
