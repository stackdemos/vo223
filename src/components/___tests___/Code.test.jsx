import renderer from 'react-test-renderer';
import React from 'react';

import {Code} from '../Code';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <Code>
                Code example
            </Code>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
