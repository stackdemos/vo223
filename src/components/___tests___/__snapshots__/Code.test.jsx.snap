// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<code
  className="code"
>
  Code example
</code>
`;
